sql/src/main/java/io/crate/executor/transport/task/UpsertByIdTask.java

```diff
@@ -124,22 +123,17 @@ public void execute(final RowReceiver rowReceiver, Row parameters) {
                 return;
             }
         }
-        Futures.addCallback(result, new FutureCallback<Long>() {
-            @Override
-            public void onSuccess(@Nullable Long result) {
-                rowReceiver.setNextRow(new Row1(result));
+        result.whenComplete(FutureCompleteConsumer.build(
```

Comment:
Why do we need to use FutureCompleteConsumer here instead of just using a lambda? This is now creating a FutureCompleteConsumer instance which still requires two lambda definitions